# HG changeset patch
# User dsamersoff
# Date 1304097180 -14400
# Node ID dddc5753c53a063c94c9949c623babe4f946c7e6
# Parent  0cddebc420d8b0fff85f4e0071d531e966372b80
7019808: build fails on Fedora 14: SELinux run-time check: execution of stack in libjvm.so
Summary: executable flag is set in the elf header of libjvm.so during build, instruct ld to don't do it.
Reviewed-by: acorn, phh

diff -r 0cddebc420d8 -r dddc5753c53a make/linux/makefiles/vm.make
--- openjdk/hotspot/make/linux/makefiles/vm.make	Thu Apr 28 08:24:46 2011 -0700
+++ openjdk/hotspot/make/linux/makefiles/vm.make	Fri Apr 29 21:13:00 2011 +0400
@@ -102,6 +102,10 @@
 CFLAGS += $(EXTRA_CFLAGS)
 LFLAGS += $(EXTRA_CFLAGS)
 
+# Don't set excutable bit on stack segment
+# the same could be done by separate execstack command
+LFLAGS += -Xlinker -z -Xlinker noexecstack
+
 LIBS += -lm -ldl -lpthread
 
 # By default, link the *.o into the library, not the executable.
